File: 045_jump_game_II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        if (nums.empty()) return 0;
        if (nums[0]==0) return 0;
        int cur=0;
        int start=-1;
        int end=0;
        if (nums.size()==1 && nums[0]) return 0;
        while (start!=end) {
            if (end>=nums.size()-1) return cur;
            cur++;
            int temp=end;
            for (int i=temp;i>start;i--) {
                if (i+nums[i]>end) end=i+nums[i];
            }
            start=temp;
       
        }
        return cur;}
};

File: algorithms/python3/045_jump_game_ii.py
class Solution:
    def jump(self, nums):
        if (len(nums)<=1):
            return 0;
        start,cur,end=-1,0,0
        while start !=end:
            if end>=len(nums)-1:
                return cur
            temp=end
            cur+=1

            for i in range(temp,start,-1):
                if i+nums[i]>end:
                    end=i+nums[i]
            start=temp
        return cur

a=Solution()
print(a.jump([2,3,1,1,4]))
        